File: Python/Task165.py
import math
print("Введіть коефіцієнти нерівності")
a = float(input())
b = float(input())
c = float(input())
d = b * b - 4 * c
if d < 0: #друга нерівність не має коренів
  print("x Є [{0:.2f}; +inf)".format(a))
else:
  if d == 0:
    x0 = -b/2
    if a < x0:
      print("x Є [{0:.3f}; {1:.3f}) U ({2:.3f}; +inf)"
          .format(a, x0, x0))
    elif a == x0:
      print("x Є ({0:.3f}; +inf)".format(a))
    else:
      print("x Є [{0:.3f}; +inf)".format(a))
  else:
    # вітка, що виконується лише при d > 0
    x1 = (-b - math.sqrt(d))/2;
    x2 = (-b + math.sqrt(d))/2;
    # формули для x1 та x2 обрані так, що x1 < x2
    if a < x1:
      print("x Є [{0:.3f}; {1:.3f}) U ({2:.3f}; +inf)"
          .format(a, x1, x2))
    elif a <= x2:
      print("x Є ({0:.3f}; +inf)".format(x2));
    else:
      print("x Є [{0:.3f}; +inf)".format(a));

File: Kotlin/Task165.kt
import kotlin.math.sqrt

fun main() {
    println("Введіть коефіцієнти нерівності")

    val a = readLine()!!.toDouble()
    val b = readLine()!!.toDouble()
    val c = readLine()!!.toDouble()

    val d = b * b - 4 * c

    if (d < 0) {
        // друга нерівність не має коренів
        println("x ∈ [$a; +inf)")
    } else {
        if (d == 0.0) {
            val x0 = -b / 2
            if (a < x0) {
                println("x ∈ [${"%.3f".format(a)}; ${"%.3f".format(x0)}) U (${ " %.3f".format(x0)}; +inf)")
            } else {
                if (a == x0) {
                    println("x ∈ (${ "%.3f".format(a) }; +inf)")
                } else {
                    println("x ∈ [${"%.3f".format(a)}; +inf)")
                }
            }
        } else {
            // d > 0
            val x1 = (-b - sqrt(d)) / 2
            val x2 = (-b + sqrt(d)) / 2  // тут гарантовано x1 < x2

            if (a < x1) {
                println(
                    "x ∈ [${"%.3f".format(a)}; ${"%.3f".format(x1)}) U (${ "%.3f".format(x2)}; +inf)"
                )
            } else {
                if (a <= x2) {
                    println("x ∈ (${ "%.3f".format(x2) }; +inf)")
                } else {
                    println("x ∈ [${"%.3f".format(a)}; +inf)")
                }
            }
        }
    }
}
